directory_downloader/directory_downloader.py

```diff
@@ -47,24 +49,37 @@ async def fetch_file_links(self, url: str = None, filter: str = None, extension:
                     next_url = url + href
                     if href.count("/") == 0:
                         file_name = next_url.split("/")[-1]
-                        if filter:
-                            matched = re.match(filter, file_name)
-                            is_match = bool(matched)
-                            if not is_match:
-                                continue
+                        if not self.is_valid_link(file_name, filter=filter, extensions=extensions):
+                            logging.warning(f"Skipping link : {next_url}")
+                            continue
 
-                        if extension:
-                            if not file_name.endswith(extension):
-                                continue
-
-                        logging.info(f"'{next_url}',")
+                        logging.info(f"{next_url}")
                         self.files_urls.add(next_url)
                     else:
-                        await self.fetch_file_links(next_url, filter=filter, extension=extension)
+                        await self.fetch_file_links(next_url, filter=filter, extensions=extensions)
 
         return self.files_urls
 
-    async def _get_site_content(self, url: str):
+    def is_valid_link(self, name, filter: Union[str, callable] = None, extensions: List[str] = None) -> bool:
```

Comment:
what if i passed 2 kwargs (filter and extenstions ) together
the extensions condition will never be reached